test/e2e/probing_test.go

```diff
@@ -0,0 +1,393 @@
+			Image:   "helloworld",
+		}
+		test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+		defer test.TearDown(clients, names)
+		objects, err := v1a1test.CreateRunLatestServiceReady(t, clients, &names)
+		if err != nil {
+			t.Fatalf("Failed to create Service %s: %v", names.Service, err)
+		}
+		domain = strings.SplitN(objects.Route.Status.URL.Host, ".", 2)[1]
+	}()
+
+	tlsOptions := &v1alpha3.TLSOptions{
+		Mode:              v1alpha3.TLSModeSimple,
+		PrivateKey:        "/etc/istio/ingressgateway-certs/tls.key",
+		ServerCertificate: "/etc/istio/ingressgateway-certs/tls.crt",
+	}
+
+	cases := []struct {
+		name    string
+		servers []v1alpha3.Server
+		urls    []string
+	}{{
+		name: "HTTP",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-http",
+				Number:   80,
+				Protocol: v1alpha3.ProtocolHTTP,
+			},
+		}},
+		urls: []string{"http://%s/"},
+	}, {
+		name: "HTTP2",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-http2",
+				Number:   80,
+				Protocol: v1alpha3.ProtocolHTTP2,
+			},
+		}},
+		urls: []string{"http://%s/"},
+	}, {
+		name: "HTTP custom port",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "custom-http",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTP,
+			},
+		}},
+		urls: []string{"http://%s:443/"},
+	}, {
+		name: "HTTP & HTTPS",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-http",
+				Number:   80,
+				Protocol: v1alpha3.ProtocolHTTP,
+			},
+		}, {
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-https",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}},
+		urls: []string{"http://%s/", "https://%s/"},
+	}, {
+		name: "HTTP redirect & HTTPS",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-http",
+				Number:   80,
+				Protocol: v1alpha3.ProtocolHTTP,
+			},
+		}, {
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-https",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}},
+		urls: []string{"http://%s/", "https://%s/"},
+	}, {
+		name: "HTTPS",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "standard-https",
+				Number:   443,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}},
+		urls: []string{"https://%s/"},
+	}, {
+		name: "HTTPS non standard port",
+		servers: []v1alpha3.Server{{
+			Hosts: []string{"*"},
+			Port: v1alpha3.Port{
+				Name:     "custom-https",
+				Number:   80,
+				Protocol: v1alpha3.ProtocolHTTPS,
+			},
+			TLS: tlsOptions,
+		}},
+		urls: []string{"https://%s:80/"},
+	}}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			names := test.ResourceNames{
```

Comment:
Can we create this once and reuse it?

Reply:
I prefer creating a new service for every test case than reusing it because it is simpler and avoid any issue that would come from caching.